tests/integration_tests/functional/test_snapshot_version.py

```diff
@@ -123,3 +131,108 @@ def test_restore_in_past_versions(bin_cloner_path):
 
         exit_code, _, _ = ssh_connection.execute_command("sleep 1 && sync")
         assert exit_code == 0
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_create_with_too_many_devices(test_microvm_with_ssh, network_config):
+    """Test scenario: restore in previous versions with too many devices."""
+    test_microvm = test_microvm_with_ssh
+    test_microvm.spawn()
+    # Set up a basic microVM.
+    test_microvm.basic_config()
+
+    # Add network devices on top of the already configured rootfs for a
+    # total of (`FC_VO_23_MAX_DEVICES_ATTACHED` + 1).
+    for i in range(FC_V0_23_MAX_DEVICES_ATTACHED):
+        # Create tap before configuring interface.
+        _tap, _host_ip, _guest_ip = test_microvm.ssh_network_config(
+            network_config,
+            str(i)
+        )
+    test_microvm.start()
+
+    ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
+    # Verify if guest can run commands.
+    exit_code, _, _ = ssh_connection.execute_command("sync")
+    assert exit_code == 0
+
+    # Create files for saving snapshot state and memory.
+    chroot_path = test_microvm.jailer.chroot_path()
+    snapshot_dir = os.path.join(chroot_path, "snapshot")
+    Path(snapshot_dir).mkdir(parents=True, exist_ok=True)
+    cmd = 'chown {}:{} {}'.format(test_microvm.jailer.uid,
+                                  test_microvm.jailer.gid,
+                                  snapshot_dir)
+    utils.run_cmd(cmd)
+
+    # Pause microVM for snapshot.
+    response = test_microvm.vm.patch(state='Paused')
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+
+    # Attempt to create a full snapshot with version: `0.23.0`.
+    # Firecracker v0.23 allowed a maximum of `FC_V0_23_MAX_DEVICES_ATTACHED`
+    # virtio devices at a time. The microVM has too many devices attached,
+    # so this should fail.
+    response = test_microvm.snapshot_create.put(
+        mem_file_path="/snapshot/vm.vmstate",
+        snapshot_path="/snapshot/vm.mem",
+        diff=False,
+        version="0.23.0"
+    )
+    assert test_microvm.api_session.is_status_bad_request(response.status_code)
+    assert "Too many devices attached. The maximum number allowed for " \
+           "the snapshot data version requested is 11." in response.text
```

Comment:
nit: matching on full literal error contents is annoying because it will fail on any wording change, I suggest you assert for only the critical part of the error message. Smth like:

```suggestion
    assert "Too many devices attached" in response.text
```

Reply:
Ack.